src/app/Fake.Sql.DacPac/Sql.SqlPackage.fs

```diff
@@ -0,0 +1,195 @@
+﻿namespace Fake.Sql
+
+/// Contains helpers around deploying databases.
+[<RequireQualifiedAccess>]
+module SqlPackage =
+
+    open Fake.Core
+    open Fake.IO.FileSystemOperators
+    open Fake.IO.Globbing.Operators
+    open System.IO
+    open System
+
+    /// The type of action to execute.
+    type DeployAction =
+        /// Generate and apply a synchronisation script between two databases.
+        | Deploy
+        /// Generate a SQL script to sync two databases.
+        | Script of OutputPath:string
+        /// Generate an XML report for the differences between two databases.
+        | Report of OutputPath:string
+
+    /// Configuration arguments for DacPac deploy
+    type DeployDbArgs = {
+        /// The path to SqlPackage.exe.
+        SqlPackageToolPath : string
+        /// Type of action to execute. Defaults to Deploy.
+        Action : DeployAction
+        /// Path to source (path to DACPAC or Connection String).
+        Source : string
+        /// Path to destination (path to DACPAC or Connection String).
+        Destination : string
+        /// Timeout for deploy (defaults to 120 seconds).
+        Timeout : int option
+        /// Block deployment if data loss can occur. Defaults to true.
+        BlockOnPossibleDataLoss : bool option
+        /// Drops objects in the destination that do not exist in the source. Defaults to false.
+        DropObjectsNotInSource : bool option
+        /// Recreates the database from scratch on publish (rather than an in-place update). Defaults to false.
+        RecreateDb : bool option
+        /// Additional configuration parameters required by sqlpackage.exe
+        AdditionalSqlPackageProperties : (string * string) list
+        /// SQLCMD variables
+        Variables : (string * string) list
+        ///Specifies the file path to a DAC Publish Profile. The profile defines a collection of properties and variables to use when generating outputs.
+        Profile : string }
+
+    let validPaths =
+        let getSqlVersion (path:string) = path.Split '\\' |> Array.item 3 |> int
+        let getVsVersion path = (Path.GetDirectoryName path |> DirectoryInfo).Name |> int
+        let sql = !!(Environment.ProgramFilesX86 </> @"Microsoft SQL Server\**\DAC\bin\SqlPackage.exe") |> Seq.map(fun path -> path, getSqlVersion path)
+        let vs = !!(Environment.ProgramFilesX86 </> @"Microsoft Visual Studio*\Common7\IDE\Extensions\Microsoft\SQLDB\DAC\*\SqlPackage.exe") |> Seq.map(fun path -> path, getVsVersion path)
+        let vs2017 = !!(Environment.ProgramFilesX86 </> @"Microsoft Visual Studio\**\Common7\IDE\Extensions\Microsoft\SQLDB\DAC\*\SqlPackage.exe") |> Seq.map(fun path -> path, getVsVersion path)
+
+        [ sql; vs; vs2017 ]
+        |> List.collect Seq.toList
+        |> List.sortByDescending snd
+        |> List.map fst
+
+    /// The default DacPac deployment arguments.
+    let DefaultDeploymentArgs = 
```

Comment:
Can we make that `internal`/`private`?

Reply:
We can make them private. There is alos few other things which can be made private, as there is no sense to expose them outside of the module. I will take care of this. I will create a new issue and send PR soon. I will take care aslo of putting link in `help/templates/template.cshtml`.

Thanks for your help!

Reply:
@kmadof Please merge release/next first, I already took care of most of the points because I wanted to release today and get this in as well.
I also splitted into a new module, plan is to remove / disable the old code eventually

Reply:
Ok. I merged release/next and I saw that all is done. Thanks even more!